### ISSUE TITLE
Branch stopped working "User session has not been initialized!"

### ISSUE THREAD

**Issue by SleeplessByte** [BranchMetrics/Branch-Android-SDK#92]
````
[behavior] CREATED: BranchMetricsLoggedInBehavior
I/BranchSDK﹕ Branch Warning: You are using your test app's Branch Key. Remember to change it to live Branch Key during deployment.
I/BranchSDK﹕ Branch Error: User session has not been initialized!
[branch-behavior] user is @derkasaurus/978
[branch-behavior] initializing for null intent data
I/BranchSDK﹕ Branch Error: User session has not been initialized!
````

- In `MyCustomApplicationClass` call `Branch.getAutoInstance( this );`.
- In `MyActivity` call `Branch.getInstance( Activity#getApplicationContext(), activity#getString( 0 R.string.api_branch_key ) );` or `Branch.getInstance( Activity#getApplicationContext()`
- Set identity using `Branch#setIdentity`
- Call `Branch#initSession( ( referringParams, error ) -> { ... } )`

The biggest problem with all this is that initSession does not return. The callback is **never** called. No error messages except for the logs. Short url generation also doesn't work.

**Comment by aaustin** [comment]
@SleeplessByte you're calling setIdentity before initSession? Is initSession in the callback?

What changed here?

**Comment by sojanpr** [comment]
@SleeplessByte  I think issue is because of trying to execute one request while another Install/open is being executed. This fix is coming with PR#76. At the time of request creation, if session is not initialized we are adding a new open/install request before. 
    Just out of curiosity, did you get the install callback after set identity failed?

**Comment by SleeplessByte** [comment]
Nope. Never get any callback (just the memory leak which is fixed in #76).

@aaustin Yes, calling `setIdentity` before `initSession`.
`initSession` is not in the callback (the callback didn't exist when I wrote this code) but the 2 requests should be queued and not "race condition".

The biggest problem is that we never get a result from the `initSession` call (or any other call). If the documentation says that the callback always be called, it **MUST**.

**Comment by SleeplessByte** [comment]
@aaustin I rewrote the piece of code so it calls `initSession` in the callback of `setIdentity`. 
The results are the same, with a more information in #94. Troubling!

**Comment by aaustin** [comment]
And you're using the latest version?

**Comment by SleeplessByte** [comment]
1.5.9; which is what we have live in the store. I think it worked before.

**Comment by SleeplessByte** [comment]
Here are the logs for 1.5.10

````
I/BranchSDK﹕ Branch Warning: You are using your test app's Branch Key. Remember to change it to live Branch Key during deployment.
I/BranchSDK﹕ Branch Error: User session has not been initialized!
[branch-behavior] ACTION: INIT SESSION null
[branch] started: true
[branch-behavior] branch identity result 
{
    "error_message": "Trouble reaching server. Please try again in a few minutes"
} 

Trouble setting the user alias. Check network connectivity and that you properly initialized
I/BranchMetricsBehavior﹕ [branch-behavior] initializing for null
I/BranchSDK﹕ Branch Error: User session has not been initialized!
````
- `Branch.getAutoInstance`
- `Branch.getInstance`
- `Branch#setIdentity`
- `Branch#initSession`

The `initSession` call is in the `setIdentity` callback. The `null` you see in the log is the intent data being null (which is correct).

**Comment by aaustin** [comment]
Do you have the Branch key in your manifest?

**Comment by aaustin** [comment]
https://dev.branch.io/recipes/quickstart_guide/android/#add-your-branch-key

**Comment by SleeplessByte** [comment]
Off course

````
<meta-data
            android:name="io.branch.sdk.BranchKey"
            android:value="@string/api_branch_key" />
````

That said, it would show
````
Log.i("BranchSDK", "Branch Warning: Please enter your branch_key in your project\'s Manifest file!");
````
if the key was missing

**Comment by aaustin** [comment]
@sojanpr is recreating your env now

**Comment by SleeplessByte** [comment]
@aaustin @sojanpr Cool. I can be there in 2 minutes if needed.

**Comment by aaustin** [comment]
That would probably be fastest :)

**Comment by sojanpr** [comment]
There was an issue of request getting stuck in the queue. Thanks to @SleeplessByte for pointing out exact issue and solution. Fix is available with v1.6.0. please update  to v1.6.0
